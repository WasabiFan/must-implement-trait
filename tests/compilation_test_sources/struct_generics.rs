use must_implement_trait::must_implement_trait;

fn main() {
    let _instance = MySendStruct { value: 7 };
}

#[must_implement_trait(Send)]
struct MySendStruct<T: Send + Sync> {
    value: T
}
