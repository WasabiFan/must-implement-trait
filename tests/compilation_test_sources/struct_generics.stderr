error: must_implement_trait does not currently support types with generic parameters.
 --> $DIR/struct_generics.rs:7:1
  |
7 | #[must_implement_trait(Send)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0422]: cannot find struct, variant or union type `MySendStruct` in this scope
 --> $DIR/struct_generics.rs:4:21
  |
4 |     let _instance = MySendStruct { value: 7 };
  |                     ^^^^^^^^^^^^ not found in this scope
